rust_src/src/keyboard.rs

```diff
@@ -271,4 +272,35 @@ pub fn command_error_default_function(
     }
 }
 
+pub const READABLE_EVENTS_DO_TIMERS_NOW: i32 = 1;
+pub const READABLE_EVENTS_FILTER_EVENTS: i32 = 2;
+
+/// Return t if command input is currently available with no wait.
+/// Actually, the value is nil only if we can be sure that no input is available;
+/// if there is a doubt, the value is t.
+///
+/// If CHECK-TIMERS is non-nil, timers that are ready to run will do so.
+#[lisp_fn(min = "0")]
+pub fn input_pending_p(check_timers: LispObject) -> bool {
+    unsafe {
+        if globals.Vunread_command_events.is_cons()
+            || globals.Vunread_post_input_method_events.is_not_nil()
+            || globals.Vunread_input_method_events.is_not_nil()
+        {
+            return true;
+        }
+
+        // Process non-user-visible events (Bug#10195).
+        process_special_events();
+
+        let val = if check_timers.is_nil() {
+            0
+        } else {
+            READABLE_EVENTS_DO_TIMERS_NOW
+        } | READABLE_EVENTS_FILTER_EVENTS;
+
+        get_input_pending(val)
```

Comment:
`get_input_pending(val | READABLE_EVENTS_FILTER_EVENTS)`
this would be a bit more readable.